guide/interface-components/components.md

```diff
@@ -0,0 +1,316 @@
+			.setStyle(ButtonStyle.Secondary);
+
+		const confirm = new ButtonBuilder()
+			.setCustomId('confirm')
+			.setLabel('Confirm')
+			.setStyle(ButtonStyle.Danger);
+
+		const row = new ActionRowBuilder()
+			.addComponents(cancel, confirm);
+
+		interaction.reply({
+			content: `Are you sure you want to ban ${target.username}?`,
+			components: [row],
+		});
+	},
+};
+```
+
+Restart your bot and test your ban command. If all goes well, you should see something like this.
+
+<DiscordMessages>
+	<DiscordMessage profile="bot">
+		<template #interactions>
+			<DiscordInteraction profile="user" :command="true">button</DiscordInteraction>
+		</template>
+		Are you sure you want to ban DiscordTroll69?
+		<template #actions>
+			<DiscordButtons>
+				<DiscordButton type="secondary">Cancel</DiscordButton>
+			</DiscordButtons>
+			<DiscordButtons>
+				<DiscordButton type="danger">Confirm</DiscordButton>
+			</DiscordButtons>
+		</template>
+	</DiscordMessage>
+</DiscordMessages>
+
+::: warning
+If you're using TypeScript you'll need to specify the type of components your action row holds. This can be done by specifying the component builder you will add to it using a generic parameter in <DocsLink path="class/ActionRowBuilder"/>.
+
+```diff
+- new ActionRowBuilder()
++ new ActionRowBuilder<ButtonBuilder>()
+```
+:::
+
+## Receiving button interactions
+
+Our buttons created and shown to the user, but there's no functionality behind them yet. For this confirmation workflow, your bot is expecting a single `ButtonInteraction` to be received before it can continue. The best way to handle this is with the <DocsLink path="class/Message?scrollTo?=awaitMessageComponent" /> method. This method wraps a Promise around an `InteractionCollector` to return only the first interaction that meets the criteria - checkout the [full guide on Collectors]() for more information on those.
+
+```js {6,10,12,15,17-20}
+module.exports = {
+	// data: new SlashCommandBuilder()...
+	async execute(interaction) {
+		// Build our buttons and row...
+
+		const sentMessage = await interaction.reply({
+			content: `Are you sure you want to ban ${target.username}?`,
+			components: [row],
+			fetchReply: true,
+		});
+
+		const filter = i => i.user.id === interaction.user.id;
+
+		try {
+			const btnInteraction = await sentMessage.awaitMessageComponent({ filter, componentType: ComponentType.Button, time: 60000 });
+		} catch (err) {
+			interaction.editReply({
+				content: 'Ban confirmation timed out',
+				components: [],
+			});
+		}
+	},
+};
+```
+
+A few things are happening here, so lets take a look at each one in order
+
+* First, we need a reference to our reply message to attach the listener to, so we set `fetchReply: true` in our response.
+* We then create a `filter` to ensure we only capture button clicks from the same user who initiated the original interaction. For anyone else, they'll see Discord's error notification since we aren't responding to those.
```

Comment:
```suggestion
* You use the `filter` function to ensure you only capture button clicks from the same user who initiated the original interaction. For anyone else, they'll see Discord's error notification since you aren't responding to them at all.
```